src/metabase/db/data_migrations.clj

```diff
@@ -334,6 +336,113 @@
       (db/update-where! model {:collection_id nil}
         :collection_id (u/get-id new-collection)))))
 
+(defn- fix-click-through
+  "Computes the visualization settings for a dashboard card based on existing card and dashboard card linked by
+  card ([:= :dashcard.card_id :card.id]).
+
+  Updates from the old style of click through to new style (#15014). Primarily moving top level `click` and
+  `click_link_template` keys to a new map under `click_behavior` with slightly renamed keys. Similarly, under
+  `column_settings` its renaming keys from `view_as` -> `type` etc.
+
+  Before (viewing them as merged card and dashboardcard visualization_settings)
+  {...
+   click: \"link\"
+   click_link_template: \"http://localhost:3001/?year={{CREATED_AT}}&cat={{CATEGORY}}&count={{count}}\"
+   column_settings {
+    [ref [field-id 6]] {
+      {view_as \"link\"
+       link_template \"http://example.com/{{ID}}\"
+       link_text \"here's an id: {{ID}}\"
+      }
+    }
+  ...
+  }
+
+  after:
+  {...
+   click_behavior {
+      type: \"link\"
+      linkType: \"url\"
+      linkTemplate: \"http://localhost:3001/?year={{CREATED_AT}}&cat={{CATEGORY}}&count={{count}}\"
+      linkTextTemplate \"here's an id: {{ID}}\"
+    }
+  column_settings {
+    [ref [field-id 6]] {
+      click_behavior {
+        type:
+        linkType:
+        linkTemplate:
+        linkTextTemplate:
+      }
+    }
+  ...
+  }"
+  [{id :id card :card_visualization dashcard :dashcard_visualization}]
+  (let [merged                   (merge card dashcard)
+        top-level-click-behavior (when (contains? merged "click")
+                                   {"type"         (get merged "click")
+                                    "linkType"     "url"
+                                    "linkTemplate" (get merged "click_link_template")})
+        has-click-behavior?      (get dashcard "click_behavior")
+        column-settings          (get merged "column_settings")
```

Comment:
I think this is wrong. If `merge` is shallow, then any dashcard column settings will shadow all card column settings, right?

Reply:
yes that's correct. is that bad? I have zero domain knowledge and just trying to glean what i can from a sql script. every conversation seems to add more domain knowledge. My guess was that the json merge in the example script was shallow as well.